go_covert_lib/controller/controller_open.go

```diff
@@ -123,6 +124,10 @@ func (ctr *Controller) retrieveChannel(cconf channelConfig) (channel.Channel, *c
 	}
 
 	switch newConf.Type {
+	case "icmpIP":
```

Comment:
As mentioned above, this will likely not compile because the icmpIP channel lacks a Receive method.